### PR TITLE
P4-1123 adding sort_by to moves#index

### PR DESCRIPTION
**Pull request by starswan**
Fixes P4-1123

## Proposed Changes

- Adds sort -> by and direction to moves index action

## To test/demo change

- test out sort parameters in postman or swagger

## Things to check & link
- [X] API docs has been updated if necessary

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -29,7 +29,10 @@ Metrics/BlockLength:
     - "spec/**/*"
 
 RSpec/NestedGroups:
-  Max: 5
+  Max: 6
+
+RSpec/MultipleExpectations:
+  Max: 2
```
Comment (alexdesi) [inline .rubocop.yml]: 👍 I like it

File: app/controllers/api/v1/moves_controller.rb [hunk 2]
```diff
@@ -4,10 +4,10 @@ module Api
   module V1
     class MovesController < ApiController
       def index
-        moves_params = Moves::ParamsValidator.new(params[:filter])
+        moves_params = Moves::ParamsValidator.new(filter_params, params[:sort] || {})
```
Comment (alexdesi) [inline app/controllers/api/v1/moves_controller.rb]: Filters and params used for the model are different entities,
so I think the validation for the `filter` should be apart from the move validation (the model).

The advantage is separation of concern and the chance to reuse the filter validator (high chance)

File: app/services/moves/finder.rb [hunk 3]
```diff
@@ -2,15 +2,40 @@
 
 module Moves
   class Finder
-    attr_accessor :filter_params, :ability
+    attr_reader :filter_params, :ability
 
-    def initialize(filter_params, ability = nil)
-      self.filter_params = filter_params
-      self.ability = ability
+    def initialize(filter_params, ability, order_params)
```
Comment (alexdesi) [inline app/services/moves/finder.rb]: we should remove the word `params` from the params :)

`def initialize(filter, ability, order)`

File: app/services/moves/finder.rb [hunk 4]
```diff
@@ -2,15 +2,40 @@
...
+        scope.left_outer_joins(:prison_transfer_reason).merge(PrisonTransferReason.ordered_by_title(@order_direction))
+      when :created_at
+        scope.order(created_at: @order_direction)
+      when :date_from
+        scope.order(date_from: @order_direction)
+      when :date
+        scope.order(date: @order_direction)
+      else
+        scope
+      end
```
Comment (alexdesi) [inline app/services/moves/finder.rb]: this method could be simplified a bit, 
maybe creating more private class with meaningful names,

File: app/services/moves/finder.rb [hunk 5]
```diff
@@ -2,15 +2,40 @@
...
+        scope.joins(:to_location).merge(Location.ordered_by_title(@order_direction))
+      when :prison_transfer_reason
+        scope.left_outer_joins(:prison_transfer_reason).merge(PrisonTransferReason.ordered_by_title(@order_direction))
+      when :created_at
+        scope.order(created_at: @order_direction)
+      when :date_from
+        scope.order(date_from: @order_direction)
+      when :date
+        scope.order(date: @order_direction)
+      else
```
Comment (alexdesi) [inline app/services/moves/finder.rb]: this seems to be the same as:
```
when [:created_at, :date_from, :date]
        scope.order(@order_by => @order_direction)
```

File: app/services/moves/params_validator.rb [hunk 6]
```diff
@@ -4,19 +4,26 @@ module Moves
   class ParamsValidator
     include ActiveModel::Validations
 
-    attr_accessor :date_from, :date_to
+    attr_reader :date_from, :date_to, :sort_by, :sort_direction
```
Comment (alexdesi) [inline app/services/moves/params_validator.rb]: `sort_by, :sort_direction` should not be here imo

File: app/controllers/api/v1/moves_controller.rb (hunk 2 above)
Comment (starswan) [reply app/controllers/api/v1/moves_controller.rb]: YAGNI

**Pull request merged**